### PR TITLE
add support for RTREE index

### PR DESCRIPTION
**Pull request by leiysky**
### What problem does this PR solve? 
Add support for RTREE index.

### What is changed and how it works?
MySQL syntax:
```
index_type:
          BTREE_SYM { $$= HA_KEY_ALG_BTREE; }
        | RTREE_SYM { $$= HA_KEY_ALG_RTREE; }
        | HASH_SYM  { $$= HA_KEY_ALG_HASH; }
        ;
```

### Check List 

Tests 

 - Unit test

### CODE REVIEW

File: parser_test.go [hunk 1]
```diff
@@ -2113,6 +2113,7 @@ func (s *testParserSuite) TestDDL(c *C) {
 		{"ALTER TABLE t ADD FULLTEXT INDEX `FullText` (`name` ASC)", true, "ALTER TABLE `t` ADD FULLTEXT `FullText`(`name`)"},
 		{"ALTER TABLE t ADD INDEX (a) USING BTREE COMMENT 'a'", true, "ALTER TABLE `t` ADD INDEX(`a`) USING BTREE COMMENT 'a'"},
 		{"ALTER TABLE t ADD INDEX IF NOT EXISTS (a) USING BTREE COMMENT 'a'", true, "ALTER TABLE `t` ADD INDEX IF NOT EXISTS(`a`) USING BTREE COMMENT 'a'"},
+		{"ALTER TABLE t ADD INDEX (a) USING RTREE COMMENT 'a'", true, "ALTER TABLE `t` ADD INDEX(`a`) USING RTREE COMMENT 'a'"},
```
Comment (leoppro) [inline parser_test.go]: add following test cases please:
```
ALTER TABLE t ADD PRIMARY KEY ident USING RTREE ( a DESC , b   )
ALTER TABLE t ADD KEY USING RTREE   ( a ) 
ALTER TABLE t ADD KEY USING RTREE ( ident ASC , ident ( 123 ) )
```

Comment (leiysky) [reply parser_test.go]: PTAL again

**Comment by leoppro** [comment]
LGTM

**Comment by leoppro** [comment]
@kennytm  PTAL

**Comment by tangenta** [comment]
LGTM

**Pull request merged**